rabbitmq/src/urabbitmq/impl/amqp_connection_handler.cpp

```diff
@@ -45,14 +46,54 @@ engine::io::Socket CreateSocket(clients::dns::Resolver& resolver,
 
 std::unique_ptr<engine::io::RwBase> CreateSocketPtr(
     clients::dns::Resolver& resolver, const AMQP::Address& address,
-    engine::Deadline deadline) {
+    const AuthSettings& auth_settings, engine::Deadline deadline) {
   auto socket = CreateSocket(resolver, address, deadline);
 
   const bool secure = address.secure();
   if (secure) {
+    if (!auth_settings.client_cert_path.empty() ||
+        !auth_settings.client_private_key_path.empty() ||
+        !auth_settings.ca_cert_paths.empty()) {
+      if (auth_settings.client_cert_path.empty() !=
```

Comment:
I feel like this validation belongs to where the settings are parsed

Reply:
Also, is there a way to express this condition in the type system?
Say, something like `std::optional<CertSettings>`

Reply:
Done